### PR TITLE
Add models for mark and sweep

### PR DESCRIPTION
**Pull request by Ladas**
Add models for mark and sweep and for sweeping job, we are allowing to re-queue it, if all the parts of the refresh were not finished.

### CODE REVIEW

File: db/migrate/20181114094606_add_last_seen_at_to_all_tables.rb [hunk 1]
```diff
@@ -0,0 +1,36 @@
+class AddLastSeenAtToAllTables < ActiveRecord::Migration[5.1]
+  def change
+    add_column :container_groups, :last_seen_at, :datetime
+    add_index :container_groups, :last_seen_at
```
Comment (agrare) [inline db/migrate/20181114094606_add_last_seen_at_to_all_tables.rb]: Pretty sure you can just pass `:index => true` to `add_column`

**Comment by agrare** [comment]
@Ladas are we going to clear these states after a successful sweep?  These seem like they could build up really quickly.

**Comment by Ladas** [comment]
@agrare I think we'll just connect it to the cleanup worker (cleaning up also old archived items, etc.)

Cause at least for the beginning, this actually shows us if all went fine (before we have a common logging in and nice stats from that)

File: db/migrate/20181114094606_add_last_seen_at_to_all_tables.rb (hunk 1 above)
Comment (Ladas) [reply db/migrate/20181114094606_add_last_seen_at_to_all_tables.rb]: ah, looks like you can't do that :-)

**Comment by Ladas** [comment]
@agrare hm, I could probably delete the :finished parts, as part of sweeping

File: db/migrate/20181114094606_add_last_seen_at_to_all_tables.rb (hunk 1 above)
Comment (agrare) [reply db/migrate/20181114094606_add_last_seen_at_to_all_tables.rb]: Hm okay, we do it here https://github.com/ManageIQ/manageiq-schema/blob/master/db/migrate/20180920085721_add_maintenance_zone_id_to_region.rb#L3
:worried:

Comment (Ladas) [reply db/migrate/20181114094606_add_last_seen_at_to_all_tables.rb]: ah, right, yeah it's funky, it doesn't throw exception, but it doesn't add the index either :-)

**Pull request merged**